### DISCUSSION TITLE
How to access some styles/resources?

### DISCUSSION THREAD

**Discussion by ghost1372** [CommunityToolkit/Labs-Windows#422] [Q&A] [answered]
### Describe the bug

After installing the nuget package, we can easily use the controls, Some controls have resources that I sometimes need to use them.
for example i want to use `SettingsCardWrapThreshold`:

`<control width="{StaticResource SettingsCardWrapThreshold}"/>`

and here we will get Error:
```
WinRT information: Cannot find a Resource with the Name/Key SettingsCardActionIconMaxSize [Line: 14 Position: 91]
XAML parsing failed.

```

is this a normal behavior? should we report this in wasdk repo?

### Steps to reproduce

```text
1. create a new project
2. install some labs components
3. use some resources
```

### Expected behavior

resources should be find and used

### Code Platform

- [ ] UWP
- [X] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [X] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Visual Studio Version

2022

### Device form factor

Desktop

### Additional context

i know that if we use Generic.xaml resource dictionary in our app we can access resources. but in Labs components we dont need to add Generic.xaml in our app.xaml file. so we should access resources

### Help us help you

No, I'm unable to contribute a solution.

**Comment by niels9001** [comment, marked as answer]
@ghost1372 If you want to get styles or resources defined in other packages, you can do that by adding this to your `Page` or `App` resources. So for the resources defined in the `SettingsCard`, that would look like this:

```xml
<Page.Resources>
    <ResourceDictionary>
        <ResourceDictionary.MergedDictionaries>
            <ResourceDictionary Source="ms-appx:///CommunityToolkit.Labs.WinUI.SettingsControls/SettingsCard/SettingsCard.xaml" />
        </ResourceDictionary.MergedDictionaries>
    </ResourceDictionary>
</Page.Resources>
```




See an example here in the `SegmentedControl` sample:

https://github.com/CommunityToolkit/Labs-Windows/blob/2eaaa484f0f9194c2ebdba455e6057cb597399f8/components/SegmentedControl/samples/SegmentedControlStylesSample.xaml#L13

Let me know if that works.

**Comment by ghost1372** [comment]
i know that i can add resource dictionary in app.xaml or page, i just want to know is this normal? so generic.xaml only works for custom controls styles and we cant use other styles/resources (we need to add resource dictionary for using styles), is it true?

**Reply by michael-hawker** [reply]
This is a limitation in how UWP XAML works compared to WPF. This is why you see the `XamlControlResources` required in the App.xaml for WinUI 2. We just don't require something similar for our library, but it does mean you need to manually import the dictionary if you want to reference any style.

This is due how to UWP XAML looks up resources, [see doc here](https://learn.microsoft.com/windows/apps/design/style/xaml-resource-dictionary#lookup-behavior-for-xaml-resource-references). It'll look into your page, then the app, and then the built-in system ones, but never look at the library's resource dictionary when coming from one of your pages.
